### PR TITLE
Make ray image parameter as optional

### PR DESCRIPTION
**Pull request by ChughShilpa**
# Issue link
https://issues.redhat.com/browse/RHOAIENG-10006 

# What changes have been made
With the new UBI9 based ray runtime image `quay.io/rhoai/ray:2.23.0-py39-cu121` used as default in codeflare-sdk raycluster-configuration, updating references of image parameter as optional in

- ClusterConfiguration docs
- demo-notebooks

# Verification steps

1. Clone the repository with git clone https://github.com/ChughShilpa/codeflare-sdk
2. Checkout this PR's branch "image_parameter"
3. Run any demo-notebooks

## Checks
- [X] I've made sure the tests are passing. 
- Testing Strategy
   - [X] Unit tests
   - [X] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Review by Bobbins228 — COMMENTED** [review]
/hold
This lgtm its just we need to hold it until after release when the new SDK version includes the optional image parameter.

File: demo-notebooks/guided-demos/notebook-ex-outputs/1_cluster_job_client.ipynb
Comment (KPostOffice) [inline demo-notebooks/guided-demos/notebook-ex-outputs/1_cluster_job_client.ipynb:45]:
```suggestion
     "If you have your own Ray image which suits your purposes, specify it in image field to override the default image."
```
Small change here and elsewhere just to simplify the language

**Review by KPostOffice — COMMENTED** [review]
Small nit

**Comment by astefanutti** [comment]
/lgtm

@Bobbins228 if I understand your concern correctly, I think we should rather move users away from relying on the main branch, rather than holding changes?

**Comment by Bobbins228** [comment]
@astefanutti That is a great idea! 
We could have the notebooks branch which has the pre-release notebooks and when we release we can sync it with the main branch. 

I did have concerns about what this current flow would be like in terms of getting source code of a specific release only to find the notebooks are outdated.

Huge +1 on this. We might need to coordinate with docs when we go through with it.

**Comment by sutaakar** [comment]
thinking, would it have sense to reference users to use tags for sample notebooks?

**Comment by Bobbins228** [comment]
Im happy for this to be merged after rebasing when #605 is merged

File: src/codeflare_sdk/demo-notebooks/guided-demos/2_basic_interactive.ipynb
Comment (astefanutti) [inline src/codeflare_sdk/demo-notebooks/guided-demos/2_basic_interactive.ipynb:74]: There is a conflict left.

**Comment by astefanutti** [comment]
/unhold

#605 has been merged.

File: src/codeflare_sdk/demo-notebooks/guided-demos/2_basic_interactive.ipynb
Comment (ChughShilpa) [reply src/codeflare_sdk/demo-notebooks/guided-demos/2_basic_interactive.ipynb]: Removed the conflict

**Review by Bobbins228 — APPROVED** [review]
/approve

**Pull request merged**
